### DISCUSSION TITLE
Can Types be generated per Row not just entire Database

### DISCUSSION THREAD

**Discussion by royboy789** [supabase/supabase#11006] [Questions] [answered]
Hello! First off, loving Supabase compared to some of the other alternatives I've been learning. 

LOVE the fact you can [create types](https://supabase.com/docs/guides/api/generating-types) from the data, makes it easy.

Currently this is what a ROW looks like coming out of the table:
```
stores: {
  Row: {
    id: number
    created_at: string | null
    user_id: string
    ...
  }
}
```

Is there a way to break the row data out so I have a new Type / Interface for each row that I can use around the UI?

```
Type Store {
  id: number
  created_at: string | null
  user_id: string
  ...
}
```
and the resulting file include it:
```
stores: {
  Row: Store
}
```

As I'm learning and starting out, doing this manually, haven't seen a discussion / questions regarding if this is possible with the typegen

Thank You!

**Comment by soedirgo** [comment, marked as answer]
Hey, thanks for using Supabase! So if I get it right, you want to extract the Row type out into its own type? You should be able to do it like so:

```ts
import { Database } from './types'

type Store = Database['public']['Tables']['stores']['Row']

const someStore: Store = { id: 123, ... }
```

This seems like a very common use case, so we'll probably put this up on the docs.

**Reply by royboy789** [reply]
Yes! I broke them apart in the types file so I can import ’Store’. But this works! That would be helpful

**Reply by aloifran** [reply]
I've been working on doing the same today by following the docs. Do you have any plans to add a command that does this automatically with the Supabase CLI? It would be super nice

**Reply by soedirgo** [reply]
@aloifran that's a good suggestion - we'll probably add some of these as helper types and/or add it to the docs.
